app/src/pages/CardSubmission/SubmissionForm.tsx

```diff
@@ -2,34 +2,55 @@ import { Button, Grid } from "@material-ui/core"
 import { CurrentGameContext } from "contexts/CurrentGame"
 import { CurrentPlayerContext } from "contexts/CurrentPlayer"
 import { useSubmitCardsMutation } from "generated/graphql"
-import { cloneDeep } from "lodash"
+import { cloneDeep, filter } from "lodash"
 import { Title } from "pages/CardSubmission"
 import SubmissionCard from "pages/CardSubmission/SubmissionCard"
 import * as React from "react"
 
 function SubmissionForm(props: { onSubmit: () => void }) {
-  const DEFAULT_NUM_ENTRIES = 3
   const currentPlayer = React.useContext(CurrentPlayerContext)
   const currentGame = React.useContext(CurrentGameContext)
   const [submitCards, { called }] = useSubmitCardsMutation()
+
+  const numSubmitted = filter(
+    currentGame.cards,
+    (card) => card.player_id === currentPlayer.id
+  ).length
+
+  const numToSubmit =
+    (currentGame.num_entries_per_player &&
+      currentGame.num_entries_per_player - numSubmitted) ||
+    0
+
   const [words, setWords] = React.useState<Array<string>>(
     Array.from(
-      { length: currentGame.num_entries_per_player || DEFAULT_NUM_ENTRIES },
+      {
+        length: numToSubmit,
+      },
       () => ""
     )
   )
 
+  React.useEffect(() => {
+    setWords(
+      Array.from(
+        {
+          length: numToSubmit,
+        },
+        () => ""
```

Comment:
we could also pull from existing state (words) and merge that into this. but 🤷‍♂️ save it for the v2. accept/reject should happen relatively quickly anyways